inst/scripts/SEL3D.R

```diff
@@ -0,0 +1,108 @@
+#Pseudocode may look like actual code...please ignore...
+get_pdb_ids <- function(target_seq, start_pos = 1 , end_pos = nchar(target_seq), convert = TRUE, silent = TRUE){
+  
+  #get packages
+  # install.packages("bio3d")
+  # install.packages("RCurl")
+  # install.packages("XML")
+  # install.packages("seqinr")
+  # library(bio3d)
+  # library(RCurl)
+  # library(XML)
+  # library(seqinr)
+  
+  
+  
+  #convert sequence into AA 
+  if(convert){
+    #translate needs a vector of single chars, not a string
+    AA_seq <- paste(translate(s2c(target_seq)), collapse='')
+    end_pos <- nchar(AA_seq)
+  }else{ AA_seq <- target_seq }
+ 
+  blast_output <- blast.pdb(AA_seq)
+  
+  blast_output <- blast_output$hit.tbl
+  blast_output <- blast_output[blast_output$identity > 40, ]
+  
+  if(nrow(blast_output) == 0){
+    return(NULL)
+  }
+  
+  blast_output$tax <- NA 
+  blast_output$info <- NA
+  
+  final_output <- data.frame()
+  final_output_range <- c()
+  final_output_range_counter <- 1
+  
+  for(entry in 1:nrow(blast_output)){
+    rcsb_id <- gsub("_", ".", blast_output$pdb.id[entry], ignore.case = TRUE)
+    lookup <- xmlParse(getURL(paste("http://www.rcsb.org/pdb/rest/describeMol?structureId=", rcsb_id, sep = "")))
+    
+    #get taxonomy 
+    
+    tax <- getNodeSet(lookup, '//Taxonomy/@name')[[1]][[1]]
+    info <- getNodeSet(lookup, '//polymerDescription/@description')[[1]][[1]]
+    
+    blast_output$tax[entry] <- ifelse(length(tax) > 0, tax, NA)
+    blast_output$info[entry] <- ifelse(length(info) > 0, yes = info, no = NA)
+    
+    if(!intersecting_positions(final_output_range, blast_output$q.start[entry], blast_output$q.end[entry])){
+      if((!is.na(blast_output$tax[entry]) && blast_output$tax[entry] == "Homo sapiens") || blast_output$identity[entry] == 100){
```

Comment:
First we drop any match with ID < 40 to save runtime, then we query each of those matches for their tax and description.

We don't want to drop all the non-homo sapien matches because of the edge case where a match could have 100% amino acid ID for a non homo sapien which we want to include in the table (we discussed this in the notes section of the wikipage).  

We then find the best match that is either Homo sapiens or 100% ID.